packages/react/src/components/Tree/Tree.tsx

```diff
@@ -263,7 +263,7 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
           const isSubtree = hasSubtree(item)
           const isSubtreeOpen = isSubtree && this.isActiveItem(item['id'])
 
-          const renderedItem = TreeItem.create(item, {
+          const renderedItem: any = TreeItem.create(item, {
```

Comment:
Without `any`, a strange TS error shows up: 
```
Type 'ShorthandValue<TreeItemProps & { kind?: never; }>' is not assignable to type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)>[]'.
  Type 'string' is not assignable to type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)>[]'.ts(2322)
```
I didn't find a way to resolve it

Reply:
Looks like TS bug, I suggest to move any to other place instead

```tsx
const renderItems = (items: ShorthandCollection<TreeItemProps>) => {
      return items.reduce<React.ReactElement[]>(
        (renderedItems, item): React.ReactElement[] => {
          const itemForRender = itemsForRender[item['id']]
          const { elementRef, ...restItemForRender } = itemForRender
          const isSubtree = hasSubtree(item)
          const isSubtreeOpen = isSubtree && this.isActiveItem(item['id'])

          const renderedItem = TreeItem.create(item, {
            defaultProps: {
              className: Tree.slotClassNames.item,
              open: isSubtreeOpen,
              renderItemTitle,
              key: item['id'],
              ...restItemForRender,
            },
            overrideProps: this.handleTreeItemOverrides,
          })

          // Only need refs of the items that spawn subtrees, when they need to be focused
          // by any of their children, using Arrow Left.
          const finalRenderedItem = isSubtree ? (
            <Ref key={item['id']} innerRef={elementRef}>
              {renderedItem}
            </Ref>
          ) : (
            renderedItem
          )

          return [
            ...renderedItems,
            finalRenderedItem,
            ...[isSubtreeOpen ? renderItems(item['items']) : []] as any,
          ]
        },
        [],
      )
    }
```